### PR TITLE
Add TextureUnmanaged

### LINKED ISSUE #211: Get raylib::Texture from raylib::RenderTexture
In `raylib::RenderTexture`, the `GetTexture()` method returns a `Texture` instead of the wrapped `raylib::Texture`. This is kinda inconvenient since instead of `render_texture.GetTexture().Draw(from, to)`, you have to use 
```
DrawTexturePro(render_texture.GetTexture(),
               from,
               to,
               raylib::Vector2(0, 0),
               0.0f,
               Color{255, 255, 255, 255});
```

### LINKED ISSUE #213: Object Pointer Classes that don't Unload
When a parent object returns a child object, the child object will destroy itself. We should have a way to have the C++ wrappers in place without the child object inadvertantly destroying itself.

```cpp
raylib::Font font("myfont.ttf");
{
  raylib::Texture fontTexture = font.GetTexture();
  fontTexture.Draw();
}
// Error: The font.texture will be destroyed since the context is garbage collected above.
```

### PR DESCRIPTION
**Pull request by RobLoach**
Add a `TextureUnmanaged` class to allow using Textures without having the garbage collector collect them.

``` cpp
raylib::Font font("myfont.ttf");
{
  raylib::TextureUnmanaged texture = font.GetTexture();
  texture.Draw();
}
// No more error since we're usning a TextureUnmanaged!

// ... Or even...
font.GetTexture().Draw();
```

Fixes #211
Fixes #213

### CODE REVIEW

**Comment by Ataraxia-Mechanica** [comment]
This looks pretty cool, it allows method chaining too which is also a plus

**Comment by RobLoach** [comment]
What do you think ab ok it having TextureUjmanaged inherit  from Texture instead ifnyhe other way around?

**Comment by Ataraxia-Mechanica** [comment]
I think both way's fine

**Comment by RobLoach** [comment]
Thinking about this more, I think having this inheitance structure makes the most sense...
```
Texture
- raylib::TextureUnmanaged
  - raylib::Texture
```

Then the majority of work is in the `TextureUnmanaged` classes, and the `Texture` just adds the managed aspect, along with the deleted copy constructor, etc.

File: examples/text/text_font_filters.cpp [hunk 1]
```diff
@@ -29,15 +29,15 @@ int main(void)
 
     // Generate mipmap levels to use trilinear filtering
     // NOTE: On 2D drawing it won't be noticeable, it looks like FILTER_BILINEAR
-    GenTextureMipmaps(&font.texture);
+    font.GetTexture().GenMipmaps();
```
Comment (RobLoach) [inline examples/text/text_font_filters.cpp:32]: This change enables us to clean up some of the example code too :+1:

**Pull request merged**